rapids-cmake/cpm/versions.json

```diff
@@ -39,6 +39,11 @@
       "version" : "1.7.0",
       "git_url" : "https://github.com/NVIDIA/libcudacxx.git",
       "git_tag" : "${version}"
+    },
+    "cuco" : {
+      "version" : "0.0.1",
```

Comment:
Since we are checking out an abitrary git SHA1 we need to specify `git_shallow` to be `false`. Since we can't presume that SHA1 will always be equal to `HEAD`, a named tag or tip of a branch.

Reply:
Oh, right. I hadn't thought about how this calls through to CPM / CMake itself. 🤦‍♂️ [This comment in CPM.cmake](https://github.com/cpm-cmake/CPM.cmake/blob/a575ab57b7e7449c896ae6fed72808ab91088954/cmake/CPM.cmake#L698-L699) indicates that hashes can't be shallow cloned by CPM. That is echoed in [the CMake docs](https://cmake.org/cmake/help/latest/module/ExternalProject.html?highlight=git_shallow):

> If `GIT_SHALLOW` is enabled then `GIT_TAG` works only with branch names and tags. A commit hash is not allowed.

[Searching the CMake source for `git_clone_options`](https://gitlab.kitware.com/search?search=git_clone_options&nav_source=navbar&project_id=541&group_id=415&search_code=true&repository_ref=master) shows where this code is used.